### ISSUE TITLE
EKS Latest kernel release 5.4.238-148.346 has no headers/devel packages

### ISSUE THREAD

**Issue by renilthomas** [awslabs/amazon-eks-ami#1266]
**Environment**:
- AWS Region: us-east-1
- EKS Platform version (use `aws eks describe-cluster --name <name> --query cluster.platformVersion`): eks.6
- Kubernetes version (use `aws eks describe-cluster --name <name> --query cluster.version`): 1.23
- AMI Version: 1.23.17-20230406
- Kernel (e.g. `uname -a`):5.4.238-148.346.amzn2.x86_64
- Release information (run `cat /etc/eks/release` on a node):

```
BASE_AMI_ID="ami-099e00fe4091e48af"
BUILD_TIME="Thu Apr  6 01:36:29 UTC 2023"
BUILD_KERNEL="5.4.238-148.346.amzn2.x86_64"
ARCH="x86_64"
```

Why this kernel version does not seem to have a linux-headers or kernel-devel pacakge. Falco relies on this to build the driver.

**Comment by HenryYanTR** [comment]
Datadog agent also crashes on this EKS AMI

**Comment by ravilr** [comment]
looks like there is only `kernel-headers-5.4.238-148.347.amzn2.x86_64` and `kernel-devel-5.4.238-148.347.amzn2.x86_64` available in the rpm repo here. 

@AWS team/ @camrakin   any idea why the latest AMI release is installing `kernel-5.4.238-148.346.amzn2.x86_64` instead of `5.4.238-148.347.amzn2.x86_64`  ( .346 versus .347) ?  perhaps the versionlock of kernel is introducing this issue ?

`sudo yum versionlock delete kernel && sudo yum -y install kernel  && sudo reboot`   does align both the `kernel` and `kernel-headers` rpm to same `...347` version and unblocks the kernel module installations.

**Comment by cartermckinnon** [comment]
Sorry for the late reply; `kernel-5.4.238-155.346` had a fatal issue affecting SMB mounts in which a null pointer dereference caused a panic. As a result, this package (and the associated `header`, `devel`, etc. packages) was removed from the Amazon Linux 2 repositories. The issue has been fixed in `kernel-5.4.238-155.347`.
